src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let finalStr = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let item = this[i];
+
+      if (item === undefined || item === null) {
+        item = '';
+      }
```

Comment:
will be better to add the item only if element not undefined or null, it will make your code shorter and clearer

Reply:
thanks, fixed